hack/config/add_crds_annotations.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# exit immediately when a command fails
+set -e
+# only exit with zero if all commands of the pipeline exit successfully
+set -o pipefail
+# error on unset variables
+set -u
+
+echo "Add annotations to CRDs"
+
+rm -f "${CONFIG_DIRECTORY:-config}"/crd/bases/*.tmp
+
+files=("${CONFIG_DIRECTORY:-config}"/crd/bases/*)
+
+# Copy and package CRDs
+for file in "${files[@]}"; do
+  {
+    cat "${file}" | sed -e "/^    controller-gen.kubebuilder.io.version.*/a \\
+    operator.solr.apache.org\\/version: ${VERSION}\\
+    argocd.argoproj.io\\/sync-options: Replace=true"
+  } > "${file}.tmp" && mv "${file}.tmp" "${file}"
```

Comment:
`sed -i` would do it in-place but may not work on some OSes

Reply:
Yeah, `sed` differences between OSX and linux drive me insane, especially the `-i`, so it's not used anywhere in the solr-operator scripts. This works for now so I'm happy enough with it, we can always improve it in the future 🙂